.pre-commit-config.yaml

```diff
@@ -31,44 +31,38 @@ repos:
       - id: mixed-line-ending
       - id: check-ast
 
-  # Formatters that may modify source files automatically
-  - repo: https://github.com/astral-sh/ruff-pre-commit
-    # Keep old ruff formatting rules until other merge requests are completed
-    rev: v0.1.11
-    hooks:
-      - id: ruff-format
-        name: ruff (format)
-        args: ["."]
-        pass_filenames: false
-
   - repo: https://github.com/adamchainz/blacken-docs
     rev: 1.16.0
     hooks:
       - id: blacken-docs
         additional_dependencies: ["black==23.10.1"]
 
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.1
+    rev: v3.16.0
     hooks:
       - id: pyupgrade
         args: ["--py38-plus", "--keep-runtime-typing"]
 
   # Linters and validation
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.3.2
+    rev: v0.4.9
     hooks:
+      - id: ruff-format
+        name: ruff (format)
+        args: ["."]
+        pass_filenames: false
```

Comment:
I think removing these two lines for both lint and fmt should switch it back to the more traditional behavior of only running on modified files, which IMO is the right behavior for a pre-commit hook (esp. since the checks run on everything in CI anyway)

Reply:
concur. only lets keep it the same version that is in `pyproject.toml` for now, and then also see if there is a dependabot configuration for this file.  That is how we keep ruff updated in `pyproject.toml`. There is a PR for `ruff` currently from dependabot but i haven't merged it yet because honestly `ruff` is releasing so frequently it's annoying.  I'm slow rolling the updates here for this dependency.  Just for awareness, I have it locked to the specific patch version because if there is a bug fix that re-enables a rule it would arbitrarily cause the CI to break upon their release.  I'm not okay with that so I'll just slow roll the merges as its only a dev dependency.

Reply:
I don't think the hook version corresponds directly to the ruff version, though they may still be linked.

Not 100% sure on dependabot, but FWIW, Renovate does have experimental support (which works quite well) for keeping track of pre-commit hook versions.

I saw the comment, but I actually think (could be wrong) that having the old version of the pre-commit pinned here might be part of why people are seeing some fighting sometimes (as maybe the hook uses an internal version of `ruff`) tied to its own version?

I do see errors with the current ruff version if I run `ruff check` locally as well, though (note: using Python 3.12 vs. 3.8, but the toml config does have target version set to 3.8) - do you not see that?

Reply:
per https://github.com/python-semantic-release/python-semantic-release/pull/958#issuecomment-2170391124, adjusted to operate only on changed files. will double-check the behavior.